src/nomad_simulation_parsers/parsers/lammps/parser.py

```diff
@@ -0,0 +1,1485 @@
+        atoms_type = atoms_info[idx].get('type')
+        if atoms_type is None:
+            return default
+        if self._chemical_symbols is None:
+            return default
+
+        try:
+            atom_labels = [self._chemical_symbols[atype] for atype in atoms_type]
+        except Exception:
+            self.logger.error('Error resolving atom labels.')
+            return
+
+        return atom_labels
+
+    def get_positions(self, idx):
+        atoms_info = self.get('atoms_info')
+        if atoms_info is None:
+            return
+
+        atoms_info = atoms_info[idx]
+
+        cell = self.get('pbc_cell')
+        cell = None if cell is None else cell[idx][1]
+        if 'xs' in atoms_info and 'ys' in atoms_info and 'zs' in atoms_info:
+            if cell is None:
+                return
+            positions = np.array(
+                [atoms_info['xs'], atoms_info['ys'], atoms_info['zs']]
+            ).T
+            positions = positions * np.linalg.norm(cell, axis=1) + np.amin(cell, axis=1)
+
+        elif 'xu' in atoms_info and 'yu' in atoms_info and 'zu' in atoms_info:
+            positions = np.array(
+                [atoms_info['xu'], atoms_info['yu'], atoms_info['zu']]
+            ).T
+
+        elif 'xsu' in atoms_info and 'ysu' in atoms_info and 'zsu' in atoms_info:
+            if cell is None:
+                return
+            positions = np.array(
+                [atoms_info['xsu'], atoms_info['ysu'], atoms_info['zsu']]
+            ).T
+            positions = positions * np.linalg.norm(cell, axis=1) + np.amin(cell, axis=1)
+
+        elif 'x' in atoms_info and 'y' in atoms_info and 'z' in atoms_info:
+            positions = np.array([atoms_info['x'], atoms_info['y'], atoms_info['z']]).T
+            if 'ix' in atoms_info and 'iy' in atoms_info and 'iz' in atoms_info:
+                if cell is None:
+                    return
+                positions_img = np.array(
+                    [atoms_info['ix'], atoms_info['iy'], atoms_info['iz']]
+                ).T
+
+                positions += positions_img * np.linalg.norm(cell, axis=1)
+        else:
+            positions = None
+
+        return positions
+
+    def get_velocities(self, idx):
+        atoms_info = self.get('atoms_info')
+        if atoms_info is None:
+            return
+        atoms_info = atoms_info[idx]
+        if 'vx' not in atoms_info or 'vy' not in atoms_info or 'vz' not in atoms_info:
+            return
+
+        return np.array([atoms_info['vx'], atoms_info['vy'], atoms_info['vz']]).T
+
+    def get_forces(self, idx):
+        atoms_info = self.get('atoms_info')
+        if atoms_info is None:
+            return
+        atoms_info = atoms_info[idx]
+        if 'fx' not in atoms_info or 'fy' not in atoms_info or 'fz' not in atoms_info:
+            return
+        return np.array([atoms_info['fx'], atoms_info['fy'], atoms_info['fz']]).T
```

Comment:
I suggest having a common function for get_forces and get_velocities e.g. get_atoms_array? that take in the key either f or v.